.github/workflows/app-publish.yaml

```diff
@@ -0,0 +1,121 @@
+    if: ${{ startsWith(github.ref, 'refs/tags/v') || github.repository == 'jellyfin/jellyfin-android' }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Setup Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Set JELLYFIN_VERSION
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: echo "JELLYFIN_VERSION=$(echo ${GITHUB_REF#refs/tags/v} | tr / -)" >> $GITHUB_ENV
+
+      - name: Build Release
+        run: ./gradlew --no-daemon --build-cache assembleRelease versionTxt
+
+      - name: Split APK release types
+        run: |
+          mkdir app/build/outputs/apk/libre_release app/build/outputs/apk/proprietary_release;
+          mv app/build/outputs/apk/**/jellyfin-android-*-libre-release.apk app/build/outputs/apk/libre_release/;
+          mv app/build/outputs/apk/**/jellyfin-android-*-proprietary-release.apk app/build/outputs/apk/proprietary_release/;
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: build-publish-artifact
+          retention-days: 30
+          if-no-files-found: error
+          path: |
+            app/build/outputs/apk/libre_release/*
+            app/build/outputs/apk/proprietary_release/*
+            app/build/version.txt
+
+  publish-release:
+    runs-on: ubuntu-20.04
+    needs: build-publish
+    if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: build-publish-artifact
+
+      - name: Set JELLYFIN_VERSION
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: echo "JELLYFIN_VERSION=$(echo ${GITHUB_REF#refs/tags/v} | tr / -)" >> $GITHUB_ENV
+
+      - name: Sign Libre Release APK
+        uses: r0adkll/sign-android-release@v1
+        id: sign_libre
+        with:
+          releaseDirectory: app/build/outputs/apk/libre_release
+          signingKeyBase64: ${{ secrets.SIGNING_KEY }}
+          alias: ${{ secrets.ALIAS }}
+          keyStorePassword: ${{ secrets.KEY_STORE_PASSWORD }}
+          keyPassword: ${{ secrets.KEY_PASSWORD }}
+
+      - name: Sign Proprietary Release APK
+        uses: r0adkll/sign-android-release@v1
+        id: sign_proprietary
+        with:
+          releaseDirectory: app/build/outputs/apk/proprietary_release
+          signingKeyBase64: ${{ secrets.SIGNING_KEY }}
+          alias: ${{ secrets.ALIAS }}
+          keyStorePassword: ${{ secrets.KEY_STORE_PASSWORD }}
+          keyPassword: ${{ secrets.KEY_PASSWORD }}
+
+      - name: Upload Release Artifacts
+        uses: ncipollo/release-action@v1.8.2
```

Comment:
Can you change it to use `alexellis/upload-assets@0.2.2`? Helps with consistency between the various Android/Kotlin projects (Kotlin SDK uses that one)

Reply:
NGL, with all the various versions of Actions you can find to do one thing this srsly gets rather convoluted and difficult to track which version is 1. maintained and 2. security patched.
The reason why I went with this one was due to its wider feature set (and since the official GH `actions/upload-release-asset` was deprecated :disappointed: )

However, I agree if you use the other action in the Kotlin SDK repo I will switch to that one :wink: